test/e2e/reconciler/cache/replication_test.go

```diff
@@ -486,6 +503,14 @@ type replicateResourceScenario struct {
 	cacheKcpClusterDynamicClient kcpdynamic.ClusterInterface
 }
 
+func (b *replicateResourceScenario) CreateAdditionalResource(ctx context.Context, t *testing.T, res runtime.Object, kind string, gvr schema.GroupVersionResource) {
```

Comment:
why not to create a resource directly in a test scenario?

Reply:
Because I don't have the clusterName at this time: the workspace is created as part of the main test, and the workspace cluster name is a field of the scenario itself.

Reply:
I think you could create a `ws` directly in the test scenario (and any other resources) and then pass the name to the replication scenario. 

xref: https://github.com/kcp-dev/kcp/blob/ba4a15b3706c339d9673b59a22cf9e783fe73f47/test/e2e/reconciler/cache/replication_test.go#L337